example/python/batch-example.py

```diff
@@ -0,0 +1,183 @@
+
+
+def add_keys_and_set_quorum():
+    alice_iroha = Iroha('alice@test')
+    alice_cmds = [
+        alice_iroha.command('AddSignatory', account_id='alice@test',
+                            public_key=ic.hex_key_to_bytes(alice_public_keys[1])),
+        alice_iroha.command('SetAccountQuorum', account_id='alice@test', quorum=2)
+    ]
+    alice_tx = alice_iroha.transaction(alice_cmds)
+    ic.sign_transaction(alice_tx, alice_private_keys[0])
+    send_transaction_and_print_status(alice_tx)
+
+    bob_iroha = Iroha('bob@test')
+    bob_cmds = [
+        bob_iroha.command('AddSignatory', account_id='bob@test', public_key=ic.hex_key_to_bytes(bob_public_keys[1])),
+        bob_iroha.command('SetAccountQuorum', account_id='bob@test', quorum=2)
+    ]
+    bob_tx = bob_iroha.transaction(bob_cmds)
+    ic.sign_transaction(bob_tx, bob_private_keys[0])
+    send_transaction_and_print_status(bob_tx)
+
+
+def alice_creates_exchange_batch():
+    alice_tx = iroha.transaction(
+        [iroha.command(
+            'TransferAsset', src_account_id='alice@test', dest_account_id='bob@test', asset_id='bitcoin#test',
+            amount='1'
+        )],
+        creator_account='alice@test',
+        quorum=2
+    )
+    bob_tx = iroha.transaction(
+        [iroha.command(
+            'TransferAsset', src_account_id='bob@test', dest_account_id='alice@test', asset_id='dogecoin#test',
+            amount='2'
+        )],
+        creator_account='bob@test'
+        # we intentionally omit here bob's quorum, since alice is the originator of the exchange and in general case
+        # alice does not know bob's quorum.
+        # bob knowing onw quorum in case of accept should sign the tx using all the number of missing keys at once
+    )
+    iroha.batch(alice_tx, bob_tx, atomic=True)
+    # sign transactions only after batch meta creation
+    ic.sign_transaction(alice_tx, *alice_private_keys)
+    send_batch_and_print_status(alice_tx, bob_tx)
+
+
+def bob_accepts_exchange_request():
+    global net
+    q = ic.sign_query(
+        Iroha('bob@test').query('GetPendingTransactions'),
+        bob_private_keys[0]
+    )
+    pending_transactions = net.send_query(q)
+    for tx in pending_transactions.transactions_response.transactions:
+        if tx.payload.reduced_payload.creator_account_id == 'alice@test':
+            del tx.signatures[:]  # we need do this temporarily, otherwise accept will not reach MST engine
+        else:
+            ic.sign_transaction(tx, *bob_private_keys)
+    send_batch_and_print_status(*pending_transactions.transactions_response.transactions)
+
+
+def check_no_pending_txs():
+    print(' ~~~ No pending txs expected:')
+    print(
+        net.send_query(
+            ic.sign_query(
+                iroha.query('GetPendingTransactions', creator_account='bob@test'),
+                bob_private_keys[0]
+            )
+        )
+    )
+    print(' ~~~')
+
+
+def bob_declines_exchange_request():
```

Comment:
This function looks exactly like `bob_accepts_exchange_request` except one line. Consider adding function with a parameter and move logic there (so `bob_accepts_exchange_request` & `bob_declines_exchange_request` are just be the wrappings around the new one)

Reply:
Not sure if this is the point of example code. Having two distinct functions makes it easier to understand the use cases for starters

Reply:
They have a lot in common due to the common nature of those two actions.
For clarity, I am not going to change this part of the example.